docs/development/setup-advanced.md

```diff
@@ -136,6 +137,116 @@ Store.
    environment](../development/setup-vagrant.html#step-4-developing),
    ignoring the parts about `vagrant` (since you're not using it).
 
+## Using the Vagrant Hyper-V provider
+
+You should have [Vagrant](https://www.vagrantup.com/downloads) and
+[Hyper-V](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v)
+installed on your system. Ensure they both work as expected.
+
+**NOTE**: Hyper-V is available only on Windows Enterprise, Pro, or Education.
+
+1. Get started by cloning the repository to your local machine.
+
+   ```bash
+   git clone --config pull.rebase git@github.com:YOURUSERNAME/zulip.git
+   ```
+
+2. You will have to open up powershell with administrator right. This is not a
+   limitation of Vagrant, but of Hyper-V itself. Then create the development
+   environment setup.
+
+   ```bash
+   cd zulip/
+   vagrant up --provider=hyperv
+   ```
+
+   You should get output like this:
+
+   ```text
+   Bringing machine 'default' up with 'hyperv' provider...
+   ==> default: Verifying Hyper-V is enabled...
+   ==> default: Verifying Hyper-V is accessible...
+   <other stuff>...
+   ==> default: Waiting for the machine to report its IP address...
+       default: Timeout: 120 seconds
+       default: IP: 172.28.119.70
+   ==> default: Waiting for machine to boot. This may take a few minutes...
+       default: SSH address: 172.28.122.156
+   ==> default: Machine booted and ready!
+   ==> default: Preparing SMB shared folders...
+   Vagrant requires administrator access for pruning SMB shares and
+   may request access to complete removal of stale shares.
+   ==> default: Starting the machine...
+   <other stuff>...
+    default: Username (user[@domain]): <your-machine-username>
+    default: Password (will be hidden):
+   ```
+
+   At this point, you will be prompted for your administrator username and password.
+   This should be your administrator account and password credentials (and not your
+   Microsoft account credentials).
+
+3. Set the `EXTERNAL_HOST` environment variable.
+
+   ```bash
+   (zulip-py3-venv) vagrant@ubuntu-18:/srv/zulip$ export EXTERNAL_HOST="$(hostname -I | xargs):9991"
```

Comment:
This switches from PowerShell to the Bash prompt inside the VM. It's implicit in the change of prompt, but best to make explicit, and probably with a specific instruction for how to get to the latter. Is it just `vagrant ssh`?